packages/ui/src/atoms/Button.tsx

```diff
@@ -90,25 +91,36 @@ export interface ButtonProps
     VariantProps<typeof buttonVariants> {
   /** Apply props and styles to child component  */
   asChild?: boolean;
+  loading?: boolean;
+  loadingChildren?: React.ReactNode;
 }
 
 export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
   (props, ref) => {
     const {
       className,
+      children,
       variant = "primary",
       size = "md",
       icon = false,
+      disabled = false,
       asChild = false,
+      loading = false,
+      loadingChildren = "Loading...",
       ...rest
     } = props;
     const Comp = asChild ? Slot : "button";
     return (
       <Comp
         className={cn(buttonVariants({ variant, size, icon }), className)}
         ref={ref}
+        disabled={disabled || loading}
         {...rest}
-      />
+      >
+        {loading ? (
+          <><Loader2 className="size-3/4 animate-spin" /> {loadingChildren}</>
```

Comment:
I would use margin left instead of space character (try `ml-2` or smth like that)

Reply:
Actually, just remove space, since there's a gap :D